### PR TITLE
MOSIP- 13633 added reg-proc external stage and service

### PR DESCRIPTION
**Pull request by neosoftSatish**

### CODE REVIEW

File: registration-processor-external-stage/src/main/java/io/mosip/registrationprocessor/externalstage/stage/ExternalStage.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	/*
+	 * (non-Javadoc)
+	 *
+	 * @see io.vertx.core.AbstractVerticle#start()
+	 */
+	@Override
+	public void start() {
+
+		router.setRoute(
+				this.postUrl(vertx, MessageBusAddress.EXTERNAL_STAGE_BUS_IN, MessageBusAddress.EXTERNAL_STAGE_BUS_OUT));
```
Comment (vishwa-vyom) [inline registration-processor-external-stage/src/main/java/io/mosip/registrationprocessor/externalstage/stage/ExternalStage.java:119]: Please change this.postUrl(vertx, to this.postUrl(getVertx(),

File: registration-processor-external-integration-service/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+## Design
+[Design - Approach for External System Integration](https://github.com/mosip/registration/blob/master/design/registration-processor/Approach_for_external_system_integration.md)
+
+[Design - Approach for Adding HTTP Stage](https://github.com/mosip/registration/blob/master/design/registration-processor/Approach_for_http_integration.md)
+
+## Default Context-path and Port
+```
+server.port=8201
+server.servlet.path=/registrationprocessor/v1/eis
+```
```
Comment (vishwa-vyom) [inline registration-processor-external-integration-service/README.md:12]: This service does not need any properties from the config server ?

File: .github/workflows/push_trigger.yml [hunk 3]
```diff
@@ -57,6 +57,55 @@ jobs:
...
       run: find -name '*.jar' -executable -type f  -exec zip release.zip {} +
 
+  docker-registration-processor-external-stage:
+    needs: build
+
+    runs-on: ubuntu-latest
+    env:
+      NAMESPACE: mosipdev
+      SERVICE_NAME: registration-processor-external-stage
+      SERVICE_LOCATION: registration-processor/pre-processor/registration-processor-external-stage
```
Comment (vishwa-vyom) [inline .github/workflows/push_trigger.yml:67]: This location is not correct

File: .github/workflows/push_trigger.yml [hunk 4]
```diff
@@ -119,6 +168,55 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
 
+  docker-registration-processor-external-integration-service:
+    needs: build
+
+    runs-on: ubuntu-latest
```
Comment (vishwa-vyom) [inline .github/workflows/push_trigger.yml:174]: This section has to be like the other in this yml file, I will anyway ask Mono to check these sections once

File: .github/workflows/push_trigger.yml [hunk 5]
```diff
@@ -57,6 +57,55 @@ jobs:
     - name: Ready the springboot artifacts
       run: find -name '*.jar' -executable -type f  -exec zip release.zip {} +
 
+  docker-registration-processor-external-stage:
```
Comment (vishwa-vyom) [inline .github/workflows/push_trigger.yml:60]: Create a folder called registration-processor and keep both the projects inside that

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor-external-integration-service/README.md (hunk 2 above)
Comment (neosoftSatish) [reply registration-processor-external-integration-service/README.md:12]: checked all classes, I have not seen any properties needed in this project.

**Review by vishwa-vyom — COMMENTED** [review]
All the review comments are taken care, requesting @MonobikashDas to review the push_trigger.yml and release_trigger.yml file alone and then we are good to merge

File: .github/workflows/push_trigger.yml (hunk 5 above)
Comment (MonobikashDas) [inline .github/workflows/push_trigger.yml:60]: Hi @neosoftSatish ,
This is not correct. You are not building code. Directly building image.
Correct steps -
1. setup env/gpg key
2. Build with Maven
3. then comes - build image (which u are doing directly here)
4. push image

CC @vishwa-vyom

**Review by MonobikashDas — CHANGES_REQUESTED** [review]
follow example of "build-pre-registration-booking-service" and do the changes for registration-processor-external-stage and external-service

**Review by MonobikashDas — APPROVED** [review]

**Pull request merged**